Small code/readme cleanup + CI fix

sungam3r opened the pull request:
[UPDATE] See #87 first.

~fixes #50~
~fixes #61~
~heap allocation optimizations~
~null argument checks~
~add comments~
~formatting~
~remove MVC from example~

sungam3r commented:
I apologize for the big PR. I just wanted to look at this repository for a long time and finally I was able to do it. Also, I expect to support this project in the future, because my team uses it in the production.

sgabler-solytic commented:
@sungam3r This looks really useful. If I understand the current state of this PR correctly, I should already be able to follow your improved Readme to setup authorization for my project without any breaking changes, right?

sungam3r commented:
Well, you may try. Basically, everything should work.

iamkoch commented:
In the README.md, the call to `AddGraphQLAuth` is wrong.

You have

```
 .AddGraphQLAuth((settings, provider) =>
```

but it should be the following, including the IServiceProvider?

```
.AddGraphQLAuth((settings, provider) =>
```

joemcbride commented:
@iamkoch the README could say .AddTeddyBears() and it would not be wrong.  That is just an example of what YOU are supposed to write in your own project.  So it is whatever YOU decide it is.

sungam3r commented:
@iamkoch There are two overloads, I use overload with one parameter, but in the readme gave an example of overload with two parameters. Fixed. I updated GraphQL dependency, there were some changes in the api.

sungam3r commented:
I need to find time to get back to this PR and make sure that it is ready for merge.

joemcbride commented on README.md:
````diff
@@ -11,66 +11,84 @@ A toolset for authorizing access to graph types for [GraphQL .NET](https://githu
+
 ```csharp
-public static void AddGraphQLAuth(this IServiceCollection services)
+public static IGraphQLBuilder AddGraphQLAuth(this IGraphQLBuilder builder, Action<AuthorizationSettings, IServiceProvider> configure)
````
This changes this to be specific to the server project.  The intent was that this can be used on its own.  I would like to keep GQL-server specifics out of this.

sungam3r replied:
I understand that. This extension from Harness example which is built on top of server project. I can return the previous variant, but then it will be different from one in the example. The value of changing the type of the `this` parameter is that a more focused api is obtained which can be chained with other extensions methods from server project. Still, this is the main scenario.

joemcbride replied:
> but then it will be different from one in the example

Yes, my intent was that the Harness should not use gql-server specific classes either, as well as the README.  Using those gives the appearance that there is a dependency when there is not.

sungam3r replied:
OK. Also I will explicitly write this thought in a readme.

sungam3r replied:
Changes were reverted.

sungam3r commented:
I will make small focused PRs.

sungam3r commented:
@joemcbride  In general, the main thing that once stopped me from continuing to work on this PR is a similar package for authorization in the server project. I saw that they are very similar. I understand that second project uses a specific ASP.NET Core auth api. Nevertheless, there was a desire to understand both and, if possible, bring them to the same denominator. Ideally, I would like the solution when there would be no copied code at all between the two projects. Do you think it's worth the effort?

joemcbride commented:
@sungam3r maybe - though the intent of this project was so that it would _not_ be ASP.NET  
Core specific, so it could also be used in other non-ASP.NET projects and ASP.NET 4.5 projects.  Others essentially copied what was here, made it ASP.NET specific, and added it to the server project.

iamkoch commented:
If the sample code is invalid or doesn't compile then why bother adding
sample code? The PR looks to address inconsistencies in the sample code
borne out of changes to the main repo, so imo I'm warranted in raising a
very minor issue about its correctness.

On Wed, 27 Nov 2019, 15:09 Joe McBride, <notifications@github.com> wrote:

> @iamkoch <https://github.com/iamkoch> the README could say
> .AddTeddyBears() and it would not be wrong. That is just an example of what
> YOU are supposed to write in your own project. So it is whatever YOU decide
> it is.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/graphql-dotnet/authorization/pull/62?email_source=notifications&email_token=AAGBMDKV2WG4GPHYFAQ46B3QV2EUJA5CNFSM4IZAKZO2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFJZSTY#issuecomment-559126863>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGBMDOSQ4BCRTQOEPVZKMTQV2EUJANCNFSM4IZAKZOQ>
> .
>

sungam3r commented:
Sample code is valid and compiles. Talk about which version of the example would be better to show.

sungam3r commented:
This PR will be revised after #84

Shane32 commented on .github/workflows/test.yml:
```diff
@@ -18,6 +18,11 @@ jobs:
+          - 3.0.0
+          - 3.1.0
+          - 3.1.3
+          - 4.0.0-preview-17
```
Nice matrix testing of prior versions

sungam3r replied:
It was your change.

Shane32 replied:
Ha it must have been awhile ago. I forgot all about it.

sungam3r commented:
@Shane32 Meet our old friend here
![изображение](https://user-images.githubusercontent.com/21261007/111962861-ea7aa100-8b03-11eb-9323-c609c41e9e5d.png)

sungam3r commented:
Regarding labeler: https://github.com/graphql-dotnet/graphql-dotnet/issues/2439
